### PR TITLE
fix(stage-deploy): build main with local bob

### PR DESCRIPTION
**Pull request by caugner**
1. Fixes the `stage-deploy` by using `mdn-bob` as specified in the `package.json` / `yarn.lock`.
2. Run the workflow on pushes to `main` (continuous deployment).
3. Format the workflow file with prettier (`npx prettier -w`).

~~Depends on https://github.com/mdn/interactive-examples/pull/2388.~~

### CODE REVIEW

File: .github/workflows/stage-deploy.yml [hunk 1]
```diff
@@ -20,17 +23,16 @@ jobs:
...
                   repository: mdn/bob
-                  ref: stage
+                  ref: main
                   path: bob
 
             - name: Setup Node.js environment
               uses: actions/setup-node@v3
               with:
                   node-version: '16'
-                  cache: yarn
```
Comment (nschonni) [inline .github/workflows/stage-deploy.yml:27]:
```suggestion
           cache: npm
           cache-dependency-path: bob/package-lock.json

```

Comment (caugner) [reply .github/workflows/stage-deploy.yml:27]: Didn't work, but I'll try again. :)

Comment (caugner) [reply .github/workflows/stage-deploy.yml:27]: Looking good, thanks, must have had a typo somewhere.

File: .github/workflows/stage-deploy.yml [hunk 2]
```diff
@@ -20,25 +28,26 @@ jobs:
...
               working-directory: bob
 
             - name: Build all
               # TODO: bob uses relative paths in bundles. For now we just symlink.
               run:  |
-                  ln -s ../bob/node_modules .
-                  node ../bob/lib/mdn-bob.js
-              working-directory: examples
+                  rm -rf ./docs
+                  npm run build ../examples
```
Comment (caugner) [inline .github/workflows/stage-deploy.yml]: This doesn't actually build the examples in `../examples`, just bob's own (test) examples.

**Comment by caugner** [comment]
@NiedziolkaMichal @nschonni Feel free to merge this PR after approving. This will not only fix the `stage-deploy` workflow, which was broken for months, but also makes sure the interactive-examples on stage are built on every push to `main`.

Once this works, we can replicate this for prod, except that we would probably want to build prod only once per day.

**Comment by NiedziolkaMichal** [comment]
This looks good to me, but I am not into GitHub workflows, so it would be great if @nschonni could check it too.

**Comment by caugner** [comment]
Alternatively, @schalkneethling might be able to have a look next week. 🙏

**Review by nschonni — APPROVED** [review]
I guess this goes from using the live bob to the NPM published version, so i think it looks fine

**Comment by caugner** [comment]
Thank you @nschonni! 🙏

**Pull request merged**